service/dap/server.go

```diff
@@ -171,6 +173,29 @@ type Config struct {
 	StopTriggered chan struct{}
 }
 
+type connection struct {
+	io.ReadWriteCloser
+	closed chan struct{}
```

Comment:
Is this preferable to a simple `bool` because of the necessary "close once and only once" semantics? E.g., bool can be switched from true/false but a chan can only be closed once?

Reply:
I guess I was modeling it after stopTriggered (aka stopChan in rpccommon).

Reply:
Ack, makes sense.